Add Workers RBAC updates (#16213)
cloudflare/cloudflare-docs: src/content/docs/workers/platform/roles-permissions.mdx

---
pcx_content_type: reference
title: Roles and permissions
sidebar:
  order: 2

---

> When creating a Cloudflare account, you will be given the Super Administrator role. As a Super Administrator, you can invite members to join your account and assign them different roles.

Reviewer: ```suggestion
When you create a Cloudflare account, you are given the [Super Administrator role](/fundamentals/setup/manage-members/roles/#account-scoped-roles). As a Super Administrator, you can invite members to join your account and assign them different roles.
```

> To check the list of members in your account, or to manage roles and permissions, refer to our [Account setup](/fundamentals/setup/manage-members/) documentation.

Reviewer: ```suggestion
To check the list of members in your account, or to manage roles and permissions, refer to the [Account setup](/fundamentals/setup/manage-members/) documentation.
```

> The following roles have permissions not restricted to Cloudflare [Workers](/workers/) but to related developer platform products too. This includes [Pages](/pages/), [Durable Objects](/durable-objects/), [KV](/kv/) and [R2](/r2/). It also provides read access to Zones, [Zone Analytics](/analytics/account-and-zone-analytics/zone-analytics/) and [Page Rules](/rules/).

Reviewer: 1. This should live within the section below
2. It seems like we should be able to enumerate these more specifically — otherwise how does someone reading this know something like, one needs the `Workers Admin` permission in order to deploy a Pages project?

Author: For #2, what do you have in mind? The copy is from a pre-existing description of Cloudflare Workers Admin role and outlines all the devplat products the role impacts.

Reviewer: Right now, when I read this, I know that which Workers role I have _somehow_ impacts what I am allowed to do with Pages, KV, etc. — but it's not explicit how.

I am saying that we should go figure out specifically things like, if I have the `Workers Read` role — what does that mean for whether I can or can't deploy a Pages project?

...and enumerate them on this page.

If we can't right now, then I don't think we should include this content about Pages, KV, etc on this page. It leaves you with more questions than answers.

## Workers roles

> Only Super Administrators will be able to assign or remove the following roles from users in their account. They also have all the permissions of a Workers Admin. Scroll to the right to see a full list of permissions for each role.

Reviewer: ```suggestion
Only Super Administrators can assign or remove the following roles from users in their account.
```

For this piece:

> Super Administrators also have all the permissions of a Workers Admin.

I would suggest that Super Administrator just be listed in the table below.

> Only Super Administrators will be able to assign or remove the following roles from users in their account. They also have all the permissions of a Workers Admin. Scroll to the right to see a full list of permissions for each role.

Reviewer: > Scroll to the right to see a full list of permissions for each role.

@kodster28 should we track something internally around making it more intuitive and visible that a table that extends horizontally beyond the viewport is scrollable, so that docs authors don't need to explain it in words?

Reviewer 2: Yeah... tables are an ongoing issue that we're discussing.

I think that's one potential solution... but there's a lot of stuff in the works + mobile and tables just really don't go together.

> |                                     | View Code | View Metrics | View Logs | Manage Versions | Manage Bindings | Manage Deployments | Change Routes | Change Subdomain | Delete Worker |

Reviewer: Each of these headings having links to relevant sections of the docs would be a very nice touch

| ----------------------------------- | --------- | ------------ | --------- | --------------- | --------------- | ------------------ | ------------- | ---------------- | ------------- |
| Workers Admin                       | ✅        | ✅            | ✅        | ✅              | ✅               | ✅                 | ✅            | ✅                | ✅            |

> | Workers Versions & Deployments Edit | ✅        | ✅            | ✅        | ✅              | ✅               | ✅                 | ✅            | ❌                | ❌            |

Reviewer: I'm not sure that someone reading this page necessarily will know specifically a Version refers to. Can we link to somewhere?
